lib/puppet/transaction/event.rb

```diff
@@ -58,11 +60,13 @@ def to_data_hash
       'name' => @name,
       'status' => @status,
       'time' => @time.iso8601(9),
-      'redacted' => @redacted
+      'redacted' => @redacted,
+      'corrective_change' => @corrective_change,
     }
   end
 
   def property=(prop)
+    @property_instance = prop
```

Comment:
There might have been a reason that the property object itself was never cached previously.  Not sure if @joshcooper might know why.